### PR TITLE
Fix clang-format hook failed when using <filename>.h

### LINKED ISSUE #10: clang-format commit hook failed when using <filename>.h
# Root cause:
Using temp file will mess up clang-format IncludeCategories rules for <filename>.h

# Description:
In clang-format, IncludeCategories will automatically set the filename.h to the highest priority, which mean it will put it at the first of the include block:

> The main header for a source file automatically gets category 0. so that it is generally kept at the beginning of the #includes [link](https://releases.llvm.org/8.0.0/tools/clang/docs/ClangFormatStyleOptions.html)

But, our hook using the temp file to compare with the source file, causing this rule not work, hence given a no-fix loop for this situation:

Report by the clang-format
```
$ clang-format this_is_the_actual_filename.c
#ifdef FOOBAR
#include "this_is_the_actual_filename.h"
#include "xxx.h"
#include "xxx.h"
#include "xxx.h"

 #include <xxx.h>
 #include <xxx.h>
```

Report by the hook
```
--- .merge_file_Zzz17X	2019-09-26 21:39:09.606926346 +0800
+++ /tmp/.merge_file_Zzz17X.I2ISWZ	2019-09-26 21:39:09.650259682 +0800
@@ -1,8 +1,8 @@
 #ifdef FOOBAR
-#include "this_is_the_actual_filename.h"
 #include "xxx.h"
 #include "xxx.h"
 #include "xxx.h"
+#include "this_is_the_actual_filename.h"
 
 #include <xxx.h>
 #include <xxx.h>
```

### PR DESCRIPTION
**Pull request by mlouielu**
Fix #10

### CODE REVIEW

File: scripts/pre-commit.hook [hunk 1]
```diff
@@ -22,13 +22,13 @@ fi
 
 FILES=`git diff --cached --name-only --diff-filter=ACMR | grep -E "\.(c|cpp|h)$"`
 for FILE in $FILES; do
-    nf=`git checkout-index --temp $FILE | cut -f 1`
-    newfile=`mktemp /tmp/${nf}.XXXXXX` || exit 1
-    $CLANG_FORMAT $nf > $newfile 2>> /dev/null
-    $DIFF -u -p -B  "${nf}" "${newfile}"
+	tempdir=`mktemp -d /tmp/.git-clang-format-check.XXXXXXXXX` || exit 1
```
Comment (jserv) [inline scripts/pre-commit.hook]: Can you follow the consistent indention?

File: scripts/pre-commit.hook [hunk 2]
```diff
@@ -22,13 +22,13 @@ fi
 
 FILES=`git diff --cached --name-only --diff-filter=ACMR | grep -E "\.(c|cpp|h)$"`
 for FILE in $FILES; do
-    nf=`git checkout-index --temp $FILE | cut -f 1`
```
Comment (jserv) [inline scripts/pre-commit.hook]: Why did you ignore `git checkout-index`?

File: scripts/pre-commit.hook (hunk 1 above)
Comment (mlouielu) [reply scripts/pre-commit.hook]: Yep, indention has changed to space

File: scripts/pre-commit.hook (hunk 2 above)
Comment (mlouielu) [reply scripts/pre-commit.hook]: Fixed

**Pull request merged**